src/main/java/com/google/api/generator/gapic/model/ResourceName.java

```diff
@@ -55,6 +57,31 @@ public abstract class ResourceName {
 
   public abstract boolean isOnlyWildcard();
 
+  @Nullable
+  public String getMatchingPattern(String binding) {
```

Comment:
Same comment as last method. The logic here also looks much more complicated, so unit tests would be really helpful.

Reply:
We don't have tests for Resourcename class directly, but it all gets tested as part of bigger test picture.

Reply:
Can we please add a test class for it? This method is pretty complicated and having unit tests will not only help coverage and help us to understand it as well(easier for us to maintain in the future). It may be already tested somewhere but unit test will always fail fast, so that it's much easier to troubleshoot.